src/core/ipns/resolver.js

```diff
@@ -102,30 +84,35 @@ class IpnsResolver {
       }
 
       // continue recursively until depth equals 0
-      this.resolver(nameSegments[2], depth - 1, peerId, resolverFn, callback)
+      this.resolver(nameSegments[2], depth - 1, callback)
     })
   }
 
-  // resolve ipns entries locally using the datastore
-  _resolveLocal (name, peerId, callback) {
-    const { ipnsKey } = ipns.getIdKeys(fromB58String(name))
+  // resolve ipns entries from the provided routing
+  _resolveName (name, callback) {
+    const peerId = PeerId.createFromB58String(name)
+    const { routingKey } = ipns.getIdKeys(fromB58String(name))
```

Comment:
Can't you use the peerId here? This way you're duplicating the calculation. 

```js
const { routingKey } = ipns.getIdKeys(peerId.toBytes())
```

Reply:
Thanks, updated!